includes/class-content-distribution.php

```diff
@@ -0,0 +1,422 @@
+<?php
+/**
+ * Newspack Network Content Distribution.
+ *
+ * @package Newspack
+ */
+
+namespace Newspack_Network;
+
+use Newspack\Data_Events;
+use WP_Post;
+use WP_Error;
+
+/**
+ * Main class for content distribution
+ */
+class Content_Distribution {
+
+	/**
+	 * The post meta key for the distributed post configuration.
+	 */
+	const DISTRIBUTED_POST_META = 'newspack_network_distributed';
+
+	/**
+	 * Post meta key for the linked post containing the distributed post hash.
+	 */
+	const POST_HASH_META = 'newspack_network_post_hash';
```

Comment:
When I read post-hash I was expecting something like a hash of site-url plus post ID, so we can verify that it's the same post on both ends. Since you're only doing an ID, maybe it can be called ID? Network id? Also, you can use `uniqid()`

Reply:
I like that. Changed to `network_post_id` using an md5 of site URL and post ID in 2d3f9ca6c2dcfa6ecfcbe06db1896e0fa1c6816c